lending/sources/lending.move

```diff
@@ -0,0 +1,375 @@
+module lending::lending {
+    use std::option;
+    use std::signer;
+    use std::string::{Self, String};
+    use std::vector;
+
+    use aptos_std::math64;
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, Metadata, MintRef, BurnRef, TransferRef, FungibleStore};
+    use aptos_framework::object::{Self, Object, ExtendRef};
+    use aptos_framework::primary_fungible_store;
+
+    const CTOKEN_DECIMALS: u8 = 8;
+    const CTOKEN_NAME_PREFIX: vector<u8> = b"CToken ";
+    const CTOKEN_SYMBOL_PREFIX: vector<u8> = b"C";
+    const BPS_BASE: u64 = 10000;
+    const DEFAULT_LIQUIDATION_INCENTIVE_BPS: u64 = 50;
+    const DEFAULT_CLOSE_FACTOR_BPS: u64 = 2000;
+
+    const ERR_UNAUTHORIZED: u64 = 1;
+    const ERR_SHORTFALL: u64 = 2;
+    const ERR_NO_SHORTFALL: u64 = 3;
+    const ERR_INSUFFICIENT_BALANCE: u64 = 4;
+    const ERR_MARKET_MISMATCH: u64 = 5;
+    const ERR_REPAY_OVER: u64 = 6;
+    const ERR_LIQUIDATE_SELF: u64 = 7;
+
+    struct Lending has key {
+        liquidation_incentive_bps: u64,
+        close_factor_bps: u64,
+    }
+
+    struct Market has key {
+        extend_ref: ExtendRef,
+        ctoken_mint_ref: MintRef,
+        ctoken_burn_ref: BurnRef,
+        ctoken_transfer_ref: TransferRef,
+        total_borrow: u64
+    }
+
+    struct Vault has key {
+        collaterals: vector<Object<Market>>,
+        debts: SimpleMap<Object<Market>, u64>
+    }
+
+    fun init_module(admin: &signer) {
+        assert!(signer::address_of(admin) == @lending, ERR_UNAUTHORIZED);
+
+        move_to(admin, Lending {
+            liquidation_incentive_bps: DEFAULT_LIQUIDATION_INCENTIVE_BPS,
+            close_factor_bps: DEFAULT_CLOSE_FACTOR_BPS,
+        });
+    }
+
+    public fun init_market(admin: &signer, underlying_asset: Object<Metadata>): Object<Market> {
+        assert!(signer::address_of(admin) == @lending, ERR_UNAUTHORIZED);
+
+        let ctoken_name = ctoken_name_via_asset_name(fungible_asset::name(underlying_asset));
+        let constructor_ref = object::create_named_object(admin, *string::bytes(&ctoken_name));
```

Comment:
Market name is the same as ctoken name?

Reply:
i think both the market and the ctoken can be named as "CUSDC"